arshal_default.go

```diff
@@ -207,11 +207,11 @@ func makeStringArshaler(t reflect.Type) *arshaler {
 		}
 
 		if mo.Flags.Get(jsonflags.StringifyBoolsAndStrings) {
-			b, err := jsontext.AppendQuote(nil, s) // only fails for invalid UTF-8
-			q, _ := jsontext.AppendQuote(nil, b)   // cannot fail since b is valid UTF-8
-			if err != nil && !xe.Flags.Get(jsonflags.AllowInvalidUTF8) {
-				return newMarshalErrorBefore(enc, t, err)
+			b, err := jsonwire.AppendQuote(nil, s, &mo.Flags)
+			if err != nil {
+				return newMarshalErrorBefore(enc, t, &jsontext.SyntacticError{Err: err})
 			}
+			q, _ := jsontext.AppendQuote(nil, b) // cannot fail since b is valid UTF-8
```

Comment:
nit: I personally use error checks with panics, as a sort of assertion to double check that the "can never happen" cases truly never happen. unless the case is from a trusted source directly, e.g. reading from crypto/rand which cannot fail as documented today.

Reply:
Done.